### DISCUSSION TITLE
Missing suggested dependencies

### DISCUSSION THREAD

**Discussion by eliashaeussler** [symfony/symfony#50494] [Q&A] [answered]
Hey together,

with #50078 (released with Symfony 6.3), suggestions for other Symfony components were removed from `composer.json` files. This leads to issues when checking project dependencies with [`composer-unused`](https://github.com/composer-unused/composer-unused).

Given the following scenario:

* I'm using the Dependency Injection component.
* I use yaml files for my service configuration.
* I don't use the Yaml component anywhere else in my codebase.
* I require `symfony/dependency-injection` as well as `symfony/yaml` to add support for yaml files within the DI component.

When checking my dependencies with `composer-unused` and **Symfony 6.2**, **no issues** are reported, because `symfonfy/yaml` is suggested by `symfony/dependency-injection` via its `composer.json` file. That's fine, as `composer-unused` is able to handle that.

However, for **Symfony 6.3**, the following error occurs:

> Unused packages
> ✗ symfony/yaml

(see for example: https://github.com/CPS-IT/frontend-asset-handler/actions/runs/5130068776/jobs/9228384844?pr=262)

This is because I don't explicitly use the Yaml component in my codebase, but configure my services through a yaml file which requires the Yaml component to be installed. But since it's no longer suggested by the DI component, `composer-unused` is unable to detect its usage.

My question is: What was the intention to remove the suggestions from `composer.json` files? Will there be any replacements or other hints/requirements about (potentially) necessary requirements for several concrete features used in Symfony components such as the DI component?

**Comment by MatTheCat** [comment, marked as answer]
Hi,

you simply cannot say a package is used only because it’s suggested by another. What if you migrated to PHP configuration before upgrading to Symfony 6.3? `symfony/yaml` would then be unused, but no error would be reported by `composer-unused`.

I guess you should use [its `--excludePackage` option](https://github.com/composer-unused/composer-unused#exclude-folders-and-packages), and maybe watch https://github.com/composer-unused/composer-unused/issues/235.

**Reply by eliashaeussler** [reply]
Thanks for your answer! Sounds legit.
